File: src/DrivingLessons.Application/Queries/FindPublicationHistory/ItemForFindPublicationHistoryResponse.cs
using DrivingLessons.Domain.Values;

namespace DrivingLessons.Application.Queries.FindPublicationHistory;

public class ItemForFindPublicationHistoryResponse
{
    public Guid PublicationId { get; init; }
    public DateOnly WeekStart { get; init; }
    public Guid TeacherId { get; init; }
    public string TeacherName { get; init; } = string.Empty;
    public PublicationState State { get; init; }
    public DateTimeOffset? WindowStartUtc { get; init; }
    public DateTimeOffset? WindowEndUtc { get; init; }
    public int? LatestExcelVersion { get; init; }
}

File: client/src/app/features/publications/data/item-for-find-publication-history.response.ts
import { PublicationState } from '../../../shared/models/publication-state.enum';

export interface ItemForFindPublicationHistoryResponse {
    publicationId: string;
    weekStart: string;
    teacherId: string;
    teacherName: string;
    state: PublicationState;
    windowStartUtc: string | null;
    windowEndUtc: string | null;
    latestExcelVersion: number | null;
}
